### ISSUE TITLE
Low GPU memory usage running run_video_face_detect.py

### ISSUE THREAD

**Issue by il-easteregg** [Linzaer/Ultra-Light-Fast-Generic-Face-Detector-1MB#165]
All required python components are installed, and running run_video_face_detect.py with a sample .MOV is successful (using `--net_type` RFB 640, `--input_size` 1280, `--test_device` cuda:0):
```
Inference time:  0.019005
Time: 0.071014s, Detect Objects: 24.
Inference time:  0.018004
Time: 0.069015s, Detect Objects: 23.
Inference time:  0.018004
Time: 0.069015s, Detect Objects: 23.
Inference time:  0.018004
Time: 0.071016s, Detect Objects: 21.
Inference time:  0.019005
Time: 0.073016s, Detect Objects: 24.
all face num:335
```
The code is running with ~15fps, but only uses 1GB out of 6GB of GPU memory (Win 10, GeForce GTX 1060 6GB) and after stopping going back to 0,4GB.
I know its a relatively large video resolution, but given the spare use of GPU memory I feel something is odd and more fps should be possible, using more GPU memory. Is this expected behaviour, or might I be missing something to use multiple streams? NVidia CUDA drivers are installed.

Thanks in advance for suggestions.

**Comment by Linzaer** [comment]
Using more GPU  memory is not directly related to FPS.

**Comment by il-easteregg** [comment]
@Linzaer Thanks, have to say I am new to the technical side of GPUs - sorry.

Since I was estimating the fps in my first post, I took the time and implemented a real fps count - in the end, the video got analyzed with ~11fps.
Your code is the fastest I've found with best precision - (by the way: Thank you for doing such a great job creating and maintaining it) - I went on and tried to speed it up even more: I gained another frame by implementing imutils and using its FileVideoStream, reaching ~12fps.
After more research, I couldn't find other possible optimizations - and in the end went for the "quick and dirty" solution, by only analyzing every second frame, pushing the frames to ~22fps.
Of course I will follow your git to see if you have other ideas for further optimizations to speed up the detection - perhaps I am lucky.

Sidenote: I failed at launching the ONNX versions of the detect scripts, they were all complaining about pytorchs caffe2/onnx/backend.cc, not translating op (slice, ...). I guess that's my fault, having done something wrong in setting it up?!
